tools/update-gcp-infra.sh

```diff
@@ -0,0 +1,168 @@
+    done
+
+    # Show status
+    show_infra
+}
+
+function start {
+    # Show status
+    show_infra
+
+    # Start all workers
+    for WORKER in `$GCLOUD compute instances list --filter="zone:$ZONE" --filter="name ~ worker-$INSTANCEID" --format="json(name)" | jq -r '.[] | .name'`
+    do
+        $GCLOUD compute instances start $WORKER --zone $ZONE --async
+    done
+
+    # Start server
+    $GCLOUD compute instances start turbinia-server-$INSTANCEID --zone $ZONE
+
+    # Show status
+    show_infra
+}
+
+function show_config {
+    # The container environment variables are *not* available in a structured format, only as a big string blob, hence the parsing...
+    show_infra
+    read -p 'Which instance? ' INSTANCE_NAME
+    $GCLOUD -q compute instances describe $INSTANCE_NAME  --format="json" --zone=$ZONE --flatten="metadata[]" | jq '.[].items[] | select(.value | contains("TURBINIA_CONF")) | .value' | sed -e 's/.*TURBINIA_CONF\\n.*value:\(.*\)\\n.*image.*/\1/' | xargs | base64 -d
+}
+
+function update_config {
+    show_infra
+
+    CONFIG_BASE64=`cat $CONFIG_FILE | base64 -w 0`
+    # Update server and workers wit the new configuration
+    for INSTANCE in `$GCLOUD compute instances list --filter="zone:$ZONE" --filter="name ~ $INSTANCEID" --format="json(name)" | jq -r '.[] | .name'`
+    do
+        $GCLOUD compute instances update-container $INSTANCE --zone $ZONE --container-env TURBINIA_CONF=$CONFIG_BASE64
+    done
+}
+
+function show_container_logs {
+    # TODO(rbdebeer) update this to use container names instead of IDs when the terraform scripts have been updated to non-random names.
+    show_infra
+    read -p 'Which instance? ' INSTANCE_NAME
+    $GCLOUD compute ssh $INSTANCE_NAME --zone us-central1-f --command="docker ps"
+    read -p 'Which container ID? ' CONTAINER_ID
+    $GCLOUD compute ssh $INSTANCE_NAME --zone us-central1-f --command="docker logs $CONTAINER_ID"
+}
+
+function update_docker_image_tag {
+    for INSTANCE in `$GCLOUD compute instances list --filter="zone:$ZONE" --filter="name ~ worker-$INSTANCEID" --format="json(name)" | jq -r '.[] | .name'`
+    do
+        $GCLOUD beta compute instances update-container $INSTANCE --zone $ZONE --container-image=$SERVER_URI:$DOCKER_TAG
+    done
+
+    for INSTANCE in `$GCLOUD compute instances list --filter="zone:$ZONE" --filter="name ~ server-$INSTANCEID" --format="json(name)" | jq -r '.[] | .name'`
+    do
+        $GCLOUD beta compute instances update-container $INSTANCE --zone $ZONE --container-image=$WORKER_URI:$DOCKER_TAG
+    done
+}
+
+# if less than 3 arguments supplied, display usage
+if [  $# -le 2 ]
+then
+    usage
+    exit 1
+fi
+
+# check whether user had supplied -h or --help . If yes display usage
+if [[ ( $# == "--help") ||  $# == "-h" ]]
+then
+    usage
+    exit 0
+fi
+
+# check if the gcloud binary is present
+check_gcloud
+
+echo "Running against GCP project:"
+$GCLOUD config list project
```

Comment:
Should we verify that this has something set?  Maybe gcloud gives an error return code if the variable isn't set?

Reply:
If no project is set it will give you the default project you authenticated against. If you did not authenticate it will show you and ask you to authenticate.